pkg/cmd/server/origin/run_components.go

```diff
@@ -265,9 +268,15 @@ func (c *MasterConfig) RunDeploymentController() {
 	if err != nil {
 		glog.Fatalf("Unable to initialize deployment controller: %v", err)
 	}
+
+	port, err := kubeserver.GetKubernetesMasterPort(&c.Options)
+	if err != nil {
+		glog.Fatal(err)
+	}
+	master := fmt.Sprintf("https://kubernetes.default.svc.%s:%d", c.Options.DNSConfig.Domain, port)
```

Comment:
the bind port may or may not match the service port... if anything, this should probably use "https://kubernetes.default.svc:$SERVICE_PORT"